koog-ktor/src/commonMain/kotlin/ai/koog/ktor/utils/EnvConfigLoader.kt

```diff
@@ -0,0 +1,146 @@
+ *
+ * @param envConfig The application configuration that contains environment-specific properties
+ *                  for configuring Koog agents and associated integrations.
+ * @return A populated instance of [KoogAgentsConfig] with the environment-specific settings applied.
+ */
+internal fun ApplicationEnvironment.loadAgentsConfig(): KoogAgentsConfig {
+    val koogConfig = KoogAgentsConfig()
+        .openAI(config)
+        .anthropic(config)
+        .google(config)
+        .openrouter(config)
+
+    if (config.propertyOrNull("koog.ollama.enable") != null) {
+        koogConfig.ollama(config)
+    }
+
+    config.propertyOrNull("koog.llm.default")?.getString()?.let { modelIdentifier ->
+        val model = getModelFromIdentifier(modelIdentifier)
+        if (model != null) {
+            koogConfig.defaultLLM = model
+        } else {
+            // TODO should this be fatal?
+            log.warn("Could not resolve model from identifier '$modelIdentifier'")
+        }
+    }
+
+    val fallbackProviderStr = config.propertyOrNull("koog.llm.fallback.provider")?.getString()
+    val fallbackModelStr = config.propertyOrNull("koog.llm.fallback.model")?.getString()
+
+    if (fallbackProviderStr != null && fallbackModelStr != null) {
+        try {
+            val fallbackProvider = when (fallbackProviderStr.lowercase()) {
+                "openai" -> LLMProvider.OpenAI
+                "anthropic" -> LLMProvider.Anthropic
+                "google" -> LLMProvider.Google
+                "openrouter" -> LLMProvider.OpenRouter
+                "ollama" -> LLMProvider.Ollama
+                else -> throw IllegalArgumentException("Unsupported LLM provider: $fallbackProviderStr")
+            }
+
+            val fullIdentifier =
+                if (fallbackProviderStr.lowercase() == "openai" && !fallbackModelStr.contains(".")) {
+                    // For OpenAI, we need to specify a category if not provided
+                    // Default to "chat" category if not specified
+                    "$fallbackProviderStr.chat.$fallbackModelStr"
+                } else {
+                    "$fallbackProviderStr.$fallbackModelStr"
+                }
+
+            val fallbackModel = getModelFromIdentifier(fullIdentifier)
+
+            if (fallbackModel != null) {
+                if (fallbackModel.provider != fallbackProvider) {
+                    log.warn("Model provider (${fallbackModel.provider.id}) does not match specified fallback provider ($fallbackProviderStr)")
+                } else {
+                    koogConfig.llm {
+                        fallback {
+                            provider = fallbackProvider
+                            model = fallbackModel
+                        }
+                    }
+                }
+            } else {
+                log.warn("Could not resolve fallback model from identifier '$fullIdentifier'")
+            }
```

Comment:
The logic here seems unnecessarily complicated
It looks like this part could be simplified without using nested ifs